vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events_unittest.cc

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+
+#include "bat/ads/internal/base/unittest/unittest_base.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_info.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_event_info.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_event_unittest_util.h"
+
+// npm run test -- brave_unit_tests --filter=BatAds*
+
+namespace ads {
+
+class BatAdsTextEmbeddingHtmlEventsTest : public UnitTestBase {
+ protected:
+  BatAdsTextEmbeddingHtmlEventsTest() = default;
+
+  ~BatAdsTextEmbeddingHtmlEventsTest() override = default;
+};
+
+TEST_F(BatAdsTextEmbeddingHtmlEventsTest, LogEvent) {
+  // Arrange
+  ml::pipeline::TextEmbeddingInfo text_embedding_info =
+      BuildTextEmbeddingInfo();
+  const std::string embedding_formatted =
```

Comment:
`embedding_as_string`, if it formatted then we should consider renaming GetVectorAsString to FormatVector